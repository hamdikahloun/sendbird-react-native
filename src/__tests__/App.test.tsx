import React from 'react';
import {render} from '@/../testing';
import App from '../App';

describe('App', () => {
  it('should render correctly', async () => {
    const screen = render(<App />).toJSON();
    expect(screen).toMatchSnapshot();
  });
});
