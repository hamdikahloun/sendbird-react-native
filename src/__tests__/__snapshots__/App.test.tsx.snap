// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      nativeID="animatedComponent"
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        hideShadow={false}
        largeTitleHideShadow={false}
        title="Login"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "flex": 1,
              "justifyContent": "center",
              "padding": 16,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/images/logo/logo.png",
              }
            }
            style={
              Object {
                "alignSelf": "center",
                "resizeMode": "center",
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#817B8B",
                "fontSize": 32,
              }
            }
          >
            Hello,
          </Text>
          <Text
            style={
              Object {
                "color": "#0C0020",
                "fontSize": 32,
              }
            }
          >
            Please enter a nickname
          </Text>
          <TextInput
            onChangeText={[Function]}
            placeholder="Nickname"
            style={
              Object {
                "backgroundColor": "#EFEFEF",
                "borderColor": "#F9F9F9",
                "borderRadius": 10,
                "borderWidth": 1,
                "height": 50,
                "marginBottom": 24,
                "marginTop": 8,
                "padding": 10,
              }
            }
            testID="nickname"
            value=""
          />
          <BVLinearGradient
            colors={
              Array [
                "#6100FF",
                "#8940FF",
              ]
            }
            style={
              Object {
                "borderRadius": 10,
                "height": 50,
                "marginHorizontal": 16,
              }
            }
          >
            <View
              accessibilityState={
                Object {
                  "disabled": true,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                  "opacity": 1,
                }
              }
              testID="login-button"
            >
              <Text
                style={
                  Object {
                    "color": "white",
                  }
                }
                testID="login-button-text"
              >
                Login
              </Text>
            </View>
          </BVLinearGradient>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
